crates/wasmtime/src/module.rs

```diff
@@ -332,8 +336,28 @@ impl Module {
         }
     }
 
-    /// Compiles a binary-encoded WebAssembly module to an artifact usable by
-    /// Wasmtime.
+    /// Creates a new WebAssembly `Module` from the contents of the given
+    /// `file` on disk, but with assumptions that the file is 'sane'.
+    /// In other words, the file should be a binary- or text-format WebAssembly
+    /// module or a precompiled version of it generated by wasmtime.
+    ///
+    /// # Unsafety
+    ///
+    /// This function is marked as `unsafe` as it relies on [`Module::deserialize`].
```

Comment:
As a matter of style, let's avoid suggesting anything about the mental health of our users. Here's an alternative, with more context copied from the documentation for `deserialize_file`, which has the same sources of unsafety. Can you think of anything this is missing that callers need to know?

```suggestion
    /// Creates a new WebAssembly `Module` from the contents of the given `file`
    /// on disk, but with assumptions that the file is from a trusted source.
    /// The file should be a binary- or text-format WebAssembly module, or a
    /// precompiled artifact generated by the same version of Wasmtime.
    ///
    /// # Unsafety
    ///
    /// All of the reasons that [`deserialize`] is `unsafe` apply to this
    /// function as well. Arbitrary data loaded from a file may trick Wasmtime
    /// into arbitrary code execution since the contents of the file are not
    /// validated to be a valid precompiled module.
    ///
    /// [`deserialize`]: Module::deserialize
    ///
    /// Additionally though this function is also `unsafe` because the file
    /// referenced must remain unchanged and a valid precompiled module for the
    /// entire lifetime of the [`Module`] returned. Any changes to the file on
    /// disk may change future instantiations of the module to be incorrect.
    /// This is because the file is mapped into memory and lazily loaded pages
    /// reflect the current state of the file, not necessarily the origianl
    /// state of the file.
```

Reply:
Great! Didn't think the scenario that the file would be modified after mmaping. Nice catch.